mlir/lib/Target/LLVMIR/ModuleTranslation.cpp

```diff
@@ -2036,6 +2036,22 @@ LogicalResult ModuleTranslation::createCommandlineMetadata() {
   return success();
 }
 
+LogicalResult ModuleTranslation::createDependentLibrariesMetadata() {
+  if (auto dependentLibrariesAttr = mlirModule->getDiscardableAttr(
+          LLVM::LLVMDialect::getDependentLibrariesAttrName())) {
+    auto *nmd =
+        llvmModule->getOrInsertNamedMetadata("llvm.dependent-libraries");
+    llvm::LLVMContext &ctx = llvmModule->getContext();
+    for (auto lib : cast<ArrayAttr>(dependentLibrariesAttr)) {
```

Comment:
```suggestion
    for (auto libAttr : dependentLibrariesAttr.getAsRange<StringAttr>()) {
```
nit: +-This should iterate over the string attributes directly.

Reply:
```
error: no member named 'getAsRange' in 'mlir::Attribute'
```

Doesn't appear to be working

Reply:
Ah right, you probably still need to cast to an array attribute first:

`cast<ArrayAttr>(dependentLibrariesAttr).getAsRange<StringAttr>()`

Your solution works as should there be another issue. I just like iterating over the type we ultimately need.